airflow/models/__init__.py

```diff
@@ -4189,16 +4243,18 @@ def create_dagrun(self,
             conf=conf,
             state=state
         )
+        (tis, edges) = self.create_tis_and_edges(execution_date)
         session.add(run)
 
+        for ti in tis:
+            session.merge(ti)
+        for edge in edges:
+            session.merge(edge)
+
         session.commit()
 
         run.dag = self
 
-        # create the associated task instances
-        # state is None at the moment of creation
-        run.verify_integrity(session=session)
-
         run.refresh_from_db()
```

Comment:
Either pass the session to this function or, wrap everything inside a `with create_session() as session:`:
```
with create_session() as session:
        run = DagRun(
            dag_id=self.dag_id,
            run_id=run_id,
            execution_date=execution_date,
            start_date=start_date,
            external_trigger=external_trigger,
            conf=conf,
            state=state
        )
        (tis, edges) = self.create_tis_and_edges(execution_date)
        session.add(run)

        for ti in tis:
            session.merge(ti)
        for edge in edges:
            session.merge(edge)
```
This will fetch a session from the pool, do the query and return it again.